packages/jaspr_content/lib/src/page.dart

```diff
@@ -355,14 +371,33 @@ class _InheritedPage extends InheritedComponent {
   }
 }
 
-extension DataMergeExtension on Map<String, dynamic> {
-  Map<String, dynamic> merge(Map<String, dynamic> other) {
-    var merged = <String, dynamic>{};
+/// The data exposed to and used by the page, accessed with [Page.data].
+///
+/// Rather than directly modifying or replacing the underlying data,
+/// use [Page.apply] to apply changes.
+extension type PageDataMap._(Map<String, Object?> _data) implements Map<String, Object?> {
+  /// Data specific to this page, namespaced under 'page' in [data].
+  Map<String, Object?> get page => switch (_data['page']) {
+        final Map<String, Object?> nestedPageData => nestedPageData,
+        _ => const {},
+      };
+
+  /// Data declared under the 'site' key in [data],
+  /// often loaded from a `_data/site.yml` file and shared across the site.
+  Map<String, Object?> get site => switch (_data['site']) {
+        final Map<String, Object?> siteData => siteData,
+        _ => const {},
+      };
+}
+
+extension DataMergeExtension on Map<String, Object?> {
+  Map<String, Object?> merge(Map<String, Object?> other) {
```

Comment:
We could make this now directly on PageDataMap, then its no longer 'polluting' the Map type